### PR TITLE
Humanoid behaviors

### PR DESCRIPTION
**Pull request by thowell**
Adds "experimental" behaviors for the humanoid including:

- robust walking with gait options 
- handstand
- flips

Fixes quadruped hill task.

### CODE REVIEW

File: mjpc/utilities.cc [hunk 1]
```diff
@@ -585,7 +585,8 @@ mjtNum Ground(const mjModel* model, const mjData* data, const mjtNum pos[3]) {
                        bodyexclude, &geomid);
 
   if (dist < 0) {  // SHOULD NOT OCCUR
-    mju_error("no group 0 geom detected by raycast");
+    return 0.0;
+    // mju_error("no group 0 geom detected by raycast");
```
Comment (yuvaltassa) [inline mjpc/utilities.cc:589]: ummm, why?
perhaps the humanoid geom groups need to change?

File: mjpc/tasks/humanoid/gait/task.xml [hunk 2]
```diff
@@ -0,0 +1,93 @@
+<mujoco model="Humanoid Locomotion">
+  <include file="../../common.xml"/>
+  <include file="../humanoid.xml" />
+  <size memory="400K"/>
+  
+  <custom>
+    <text name="task_transition" data="Humanoid|Gait|Get Up|Flip|Hand Stand" />
```
Comment (yuvaltassa) [inline mjpc/tasks/humanoid/gait/task.xml:7]: Flip and handstand work? 

I would expect the shoulders would need to be modified for handstand?

File: mjpc/utilities.cc (hunk 1 above)
Comment (thowell) [reply mjpc/utilities.cc:589]: This method oftens errors for humanoid. Maybe I didn't set all of the geom `group` specifications correctly?

File: mjpc/tasks/humanoid/gait/task.xml (hunk 2 above)
Comment (thowell) [reply mjpc/tasks/humanoid/gait/task.xml]: These behaviors are prototyped and work with slowdown, although not with 100% reliability. 
A modified humanoid `humanoid_strong.xml` increases joint limits and gearing to achieve the behaviors. However, the dynamics become more unstable. We should probably transition to the CMU humanoid for these tasks.

**Comment by thowell** [comment]
The behaviors can still be improved, but this code is ready for review.

**Pull request closed without merging**